### PR TITLE
bug 1199166 - disallow robots from comparing

### PR DESCRIPTION
**Pull request by groovecoder**
[Today's down-time incident](https://rpm.newrelic.com/accounts/263620/applications/3172075/downtime/16882252) was [caused by Yahoo Slurp! crawling onto a bad and expensive compare transaction](https://bugzilla.mozilla.org/show_bug.cgi?id=1199166#c1).

To help mitigate the risk of search crawlers triggering overload transactions, we want to add a number of new `Disallow` directives to our `robots.txt`, so we can expect well-behaved robots to stay away from them.

I used Google Webmaster Tools to verify that this `robots.txt` file will not disallow regular docs urls. I couldn't find equivalent webmaster tools for Yahoo, so I emailed Kev Needham (our Yahoo liaison) and set up a Google Analytics alert to email me if our traffic from Yahoo drops below our minimum observed threshold, just in case.

### CODE REVIEW

File: media/robots.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-User-Agent: *
+User-agent: *
```
Comment (jwhitlock) [inline media/robots.txt]: was this a suggested change?

File: media/robots.txt [hunk 2]
```diff
@@ -6,5 +6,30 @@ Request-rate: 1/5
 Disallow: /*feed=rss
 Disallow: /*type=feed
 Disallow: /skins
-Disallow: /template:
+Disallow: /*docs/feeds
+Disallow: /*docs/templates
+Disallow: /*docs*Template
```
Comment (jwhitlock) [inline media/robots.txt]: will this block https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XUL/Tutorial/Templates?

File: media/robots.txt [hunk 3]
```diff
@@ -6,5 +6,30 @@ Request-rate: 1/5
...
+Disallow: /*docs*$move
+Disallow: /*docs*$quick-review
+Disallow: /*docs*$revert
+Disallow: /*docs*$repair_breadcrumbs
+Disallow: /*docs*$delete
+Disallow: /*docs*$restore
+Disallow: /*docs*$purge
+Disallow: /*docs*$subscribe
+Disallow: /*docs*$vote
+Disallow: /*docs*.json
```
Comment (jwhitlock) [inline media/robots.txt]: will this block https://developer.mozilla.org/en-US/Add-ons/SDK/Tools/package_json ?

File: media/robots.txt (hunk 1 above)
Comment (groovecoder) [inline media/robots.txt]: Yes, suggested by Google webmaster tools

File: media/robots.txt (hunk 2 above)
Comment (groovecoder) [inline media/robots.txt]: Good catch. Updated.

File: media/robots.txt (hunk 3 above)
Comment (groovecoder) [inline media/robots.txt]: Nope. robots.txt directives aren't regex :neutral_face: 
mdn-robots-json-okay

**Comment by jwhitlock** [comment]
+r would crawl again

**Comment by groovecoder** [comment]
giphy

**Pull request merged**